id/_internal/oidc/ambient.py

```diff
@@ -119,6 +124,7 @@ def detect_gcp(audience: str) -> Optional[str]:
             _GCP_TOKEN_REQUEST_URL,
             params={"scopes": "https://www.googleapis.com/auth/cloud-platform"},
             headers={"Metadata-Flavor": "Google"},
+            timeout=timeout,
```

Comment:
We should probably explicitly handle the consequent `Timeout` exception here and below, by turning it into one of our exception types. That we we don't leak as many third-party exception types 🙂

Reply:
Whoops, yeah. I thought that was a subclass of the one being caught, will fix this now.

Reply:
[d80a776](https://github.com/di/id/pull/26/commits/d80a776b7a9b5c68725dc5d190a25861338fbef0)